### PR TITLE
Add the consolidated release notes for 2.0.0-rc1 release.

### PR DESCRIPTION
**Pull request by zelinh**
Signed-off-by: Zelin Hao <zelinhao@amazon.com>

### Description
Add the consolidated release notes for 2.0.0-rc1 release.

### Issues Resolved
Part of #1624 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: release-notes/opensearch-release-notes-2.0.0-rc1.md [hunk 1]
```diff
@@ -0,0 +1,606 @@
...
+* Notifications consist of three plugins, `notifications-core` and `notifications` backend plugins for OpenSearch, and a `notificationsDashboards` frontend plugin for OpenSearch Dashboards.
+
+
+## Release Details
+
+OpenSearch and OpenSearch Dashboards 2.0.0-rc1 include the following features, enhancements, bug fixes, infrastructure, documentation, maintenance, and refactoring updates.
+
+OpenSearch [Release Notes](https://github.com/opensearch-project/OpenSearch/blob/main/release-notes/opensearch.release-notes-2.0.0-rc1.md).
+
+OpenSearch Dashboards [Release Notes](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/release-notes/opensearch-dashboards.release-notes-2.0.0-rc1.md).
```
Comment (zelinh) [inline release-notes/opensearch-release-notes-2.0.0-rc1.md:16]: The release notes for OpenSearch & OpenSearch Dashboards are not published yet. I put the potential links of them here.

File: release-notes/opensearch-release-notes-2.0.0-rc1.md [hunk 2]
```diff
@@ -0,0 +1,606 @@
+# OpenSearch and Dashboards 2.0.0-rc1 Release Notes
+
+## Release Highlights
```
Comment (bbarani) [inline release-notes/opensearch-release-notes-2.0.0-rc1.md:3]: Can we highlight the native KNN support and other notable Lucene 9.1 highlights ?

File: release-notes/opensearch-release-notes-2.0.0-rc1.md [hunk 3]
```diff
@@ -0,0 +1,606 @@
...
+
+### OpenSearch Notifications
+* OpenSearch 2.0.0 is the first official release with OpenSearch Notifications.
```
Comment (stockholmux) [inline release-notes/opensearch-release-notes-2.0.0-rc1.md:6]: 2.0.0 ga? or 2.0.0 rc1?

File: release-notes/opensearch-release-notes-2.0.0-rc1.md [hunk 4]
```diff
@@ -0,0 +1,606 @@
...
+
+### OpenSearch Observability
+* Modularize and cleanup traces ([#601](https://github.com/opensearch-project/observability/pull/601))
+* Modularize and cleanup panel ([#603](https://github.com/opensearch-project/observability/pull/603))
+* Modularize event Analytics live tail and fix bug ([#647](https://github.com/opensearch-project/observability/pull/647))
+* Fix lint and modularize dashboard ([#583](https://github.com/opensearch-project/observability/pull/583))
+* Modularize service and fix issues ([#595](https://github.com/opensearch-project/observability/pull/595))
+
+
+## Deprecation
```
Comment (stockholmux) [inline release-notes/opensearch-release-notes-2.0.0-rc1.md:602]: If we have this heading we should be putting the all the inclusive language stuff here too.

**Review by stockholmux — DISMISSED** [review]
1 nit and 1 major issue.

File: release-notes/opensearch-release-notes-2.0.0-rc1.md (hunk 2 above)
Comment (elfisher) [reply release-notes/opensearch-release-notes-2.0.0-rc1.md:3]: lucene-based KNN isn't in 2.0. I believe it is being tracked here: https://github.com/opensearch-project/k-NN/issues/39

@vamshin pls confirm.

Comment (elfisher) [inline release-notes/opensearch-release-notes-2.0.0-rc1.md:3]: Some release highlights we should also include are:
* Document level alerting
* Lucene 9
* geo map tile generation pipeline https://github.com/opensearch-project/maps/issues/2
* Term lookup query support in document level security https://github.com/opensearch-project/security/issues/1508

**Review by elfisher — DISMISSED** [review]
some changes requested.

File: release-notes/opensearch-release-notes-2.0.0-rc1.md (hunk 1 above)
Comment (peterzhuamazon) [reply release-notes/opensearch-release-notes-2.0.0-rc1.md:16]: OS Core: https://github.com/opensearch-project/OpenSearch/blob/2.0/release-notes/opensearch.release-notes-2.0.0-rc1.md

File: release-notes/opensearch-release-notes-2.0.0-rc1.md (hunk 2 above)
Comment (zelinh) [reply release-notes/opensearch-release-notes-2.0.0-rc1.md:3]: @elfisher Any wording suggestion for these highlights? The highlight section wasn't included in the plugin release notes.

File: release-notes/opensearch-release-notes-2.0.0-rc1.md (hunk 4 above)
Comment (zelinh) [reply release-notes/opensearch-release-notes-2.0.0-rc1.md]: This section is removed as PA moved this to `Documentation`. 
https://github.com/opensearch-project/performance-analyzer/pull/196

File: release-notes/opensearch-release-notes-2.0.0-rc1.md (hunk 2 above)
Comment (bbarani) [reply release-notes/opensearch-release-notes-2.0.0-rc1.md:3]: @CEHENKLE @davidlago Can you help us on the release highlights?

Comment (elfisher) [reply release-notes/opensearch-release-notes-2.0.0-rc1.md:3]: Please use the following.

- Document level alerting allows users to create monitors that can generate alerts per document.
- Lucene 9 is now used in OpenSearch
- The Geo Map Tiles in OpenSearch Dashboards are updated and now have a pipeline to update them more frequently.
- Document level security now supports term lookup queries.

Comment (zelinh) [reply release-notes/opensearch-release-notes-2.0.0-rc1.md:3]: Added these to the release highlights.

**Review by CEHENKLE — APPROVED** [review]

**Review by bbarani — APPROVED** [review]

**Pull request merged**